src/parquet/column/reader.h

```diff
@@ -184,7 +188,7 @@ inline int64_t TypedColumnReader<DType>::ReadBatch(int batch_size, int16_t* def_
   }
 
   // Not present for non-repeated fields
-  if (descr_->max_repetition_level() > 0) {
+  if (descr_->max_repetition_level() > 0 && rep_levels) {
     num_rep_levels = ReadRepetitionLevels(batch_size, rep_levels);
     if (num_def_levels != num_rep_levels) {
       throw ParquetException("Number of decoded rep / def levels did not match");
```

Comment:
We could certainly insert a DCHECK that verifies that if `rep_levels` is not nullptr then neither is `def_levels`. 

@edani could you explain how this issue manifested itself? how I'm reading what you said about Spark is that there are always definition levels, even when `max_definition_level` is 0. parquet-cpp can just skip decoding the data in this case.